### ISSUE TITLE
Go Mod Toolkit more specific version

### ISSUE THREAD

**Issue by Guneetsinghtuli** [influxdata/telegraf#15398]
### Relevant telegraf.conf

```toml
Not required as no config was setup
```

### Logs from Telegraf

```text
go: downloading go1.22 (darwin/arm64)
go: download go1.22 for darwin/arm64: toolchain not available
CGO_ENABLED=0 go build -tags "" -ldflags " -X github.com/influxdata/telegraf/internal.Commit=4aa288b2 -X github.com/influxdata/telegraf/internal.Branch=fix/go-toolkit -X github.com/influxdata/telegraf/internal.Version=1.31.0-4aa288b2" ./cmd/telegraf
go: downloading go1.22 (darwin/arm64)
go: download go1.22 for darwin/arm64: toolchain not available
make: *** [build] Error 1
```

### System info

Telegraf 1.20.4, Mac darwin/arm64 

### Steps to reproduce

1. On using the `1.22` Go toolkit version and running `make build` 

### Expected behavior

It should be building telegraf

### Actual behavior

Got Errored with error toolchain not available

**Comment by powersj** [comment]
Should grab this with https://github.com/influxdata/telegraf/pull/15383

**Comment by powersj** [comment]
Thanks again for the PR and issue. I've merged #15383, so please do let us know if this does not resolve the issue.
